AndroidXCI/lib/src/main/kotlin/dev/androidx/ci/testRunner/TestRunnerServiceImpl.kt

```diff
@@ -221,13 +207,44 @@ internal class TestRunnerServiceImpl internal constructor(
         }
     }
 
+    private suspend fun findScreenshotFiles(
+        resultPath: GcsPath,
+        testIdentifier: TestRunnerService.TestIdentifier
+    ): List<TestRunnerService.TestCaseArtifact> {
+        val screenshotArtifacts = mutableListOf<TestRunnerService.TestCaseArtifact>()
+        val testName = "${testIdentifier.className}_${testIdentifier.name}"
+        fun BlobVisitor.fullDeviceId() = relativePath.substringBefore('/', "")
+        googleCloudApi.walkEntires(
+            gcsPath = resultPath
+        ).forEach { visitor ->
+            val runNumber = DeviceRun.create(visitor.fullDeviceId()).runNumber
+            if (visitor.fileName.startsWith(testName) && runNumber == testIdentifier.runNumber) {
+                screenshotArtifacts.add(
+                    TestRunnerService.TestCaseArtifact(
+                        ResultFileResourceImpl(visitor),
+                        visitor.fileName.substringAfterLast(".")
+                    )
+                )
+            }
+        }
+        return screenshotArtifacts
+    }
+
     suspend fun getTestMatrixResults(
         testMatrixId: String
     ): List<TestRunnerService.TestRunResult>? {
         val testMatrix = testLabController.getTestMatrix(testMatrixId) ?: return null
         return getTestMatrixResults(testMatrix)
     }
 
+    suspend fun getTestMatrixResultsScreenshots(
```

Comment:
unless we have a way to avoid calling this for each test in a TestMatrix, this is not going to scale well because it is traversing all of GcsPath just to find 1 test's screenshots, and does the same again for another test.

If we want this API, it should get a list of test idenfieris and find all of them at once

Reply:
Updated. Please review!